pkg/plugin/datasource.go

```diff
@@ -39,8 +43,23 @@ func (instance *LynxDataSourceInstance) Dispose() {
 // QueryData handler for data queries
 func (instance *LynxDataSourceInstance) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
 	response := backend.NewQueryDataResponse()
+	var usedInstance = instance
+	if instance.options.OAuth2Passthru {
+		headers := req.GetHTTPHeaders()
+		authHeader := headers.Get(backend.OAuthIdentityTokenHeaderName)
+		parts := strings.Fields(authHeader)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("bad auth header")
+		}
+		usedInstance = &LynxDataSourceInstance{
+			client: lynx.NewClient(&lynx.Options{
+				Authenticator: lynx.AuthApiKey{Key: parts[1]},
```

Comment:
The auth here should use AuthBearer instead

Reply:
Fix'd